lib/portage/tests/resolver/test_use_dep_defaults.py

```diff
@@ -55,3 +55,195 @@ def testUseDepDefaultse(self):
+                "RDEPEND": "gnuefi? ( sys-boot/gnu-efi )",
+            },
+            "sys-kernel/installkernel-gentoo-7": {
+                "EAPI": 8,
+                "RDEPEND": "!sys-kernel/installkernel-systemd",
+            },
+        }
+        ebuilds = {
+            "sys-boot/gnu-efi-3.0.15": {},
+            "sys-apps/systemd-253.11": {
+                "EAPI": 8,
+                "IUSE": "gnuefi",
+                "RDEPEND": "gnuefi? ( sys-boot/gnu-efi )",
+            },
+            "sys-apps/systemd-254.5": {
+                "EAPI": 8,
+                "IUSE": "kernel-install",
+            },
+            "sys-kernel/installkernel-gentoo-7": {
+                "EAPI": 8,
+                "RDEPEND": "!sys-kernel/installkernel-systemd",
+            },
+            "sys-kernel/installkernel-systemd-2-r4": {
+                "EAPI": 8,
+                "RDEPEND": """
+                    !sys-kernel/installkernel-gentoo
+                    || (
+                        sys-apps/systemd[gnuefi(-)]
+                        sys-apps/systemd[kernel-install(-)]
+                    )
+                """,
+            },
+        }
+
+        test_cases = (
+            # The following USE changes are necessary to proceed:
+            # (see "package.use" in the portage(5) man page for more details)
+            # # required by sys-kernel/installkernel-systemd-2-r4::test_repo
+            # # required by sys-kernel/installkernel-systemd (argument)
+            # =sys-apps/systemd-253.11 gnuefi
+            ResolverPlaygroundTestCase(
+                ["sys-kernel/installkernel-systemd"],
+                success=False,
+                ambiguous_merge_order=True,
+                options={
+                    "--autounmask": "y",
+                },
+                mergelist=[
+                    "sys-apps/systemd-254.5",
+                    "sys-kernel/installkernel-systemd-2-r4",
+                    "[uninstall]sys-kernel/installkernel-gentoo-7",
+                    (
+                        "!sys-kernel/installkernel-gentoo",
+                        "!sys-kernel/installkernel-systemd",
+                    ),
+                ],
+                use_changes={"sys-apps/systemd": {"boot": True}},
+            ),
+            # emerge: there are no ebuilds built with USE flags to satisfy "sys-apps/systemd[gnuefi(-)]".
+            # !!! One of the following packages is required to complete your request:
+            # - sys-apps/systemd-253.11::test_repo (Change USE: +gnuefi)
+            # (dependency required by "sys-kernel/installkernel-systemd-2-r4::test_repo" [ebuild])
+            # (dependency required by "sys-kernel/installkernel-systemd" [argument])
+            ResolverPlaygroundTestCase(
+                ["sys-kernel/installkernel-systemd"],
+                success=False,
+                ambiguous_merge_order=True,
+                options={
+                    "--autounmask": "n",
```

Comment:
I'm not sure if we have a way to test for the ".. is required to complete your request" bit, so maybe I'll drop this one.